updater/lib/dependabot/updater/operations/group_update_all_versions.rb

```diff
@@ -155,7 +194,7 @@ def compile_updates_for(dependency, dependency_files)
             return []
           end
 
-          filter_unrelated_and_unchanged(updated_deps, checker)
+          updated_deps
```

Comment:
We still do perform this filtering, it just [happens inside](https://github.com/dependabot/dependabot-core/pull/7004/files#diff-a649578407b6af44340c7b21fc05ed2be2c0d5cf2e2b6a19adb363343ff57086R38-R44) the `DependencyChangeBuilder` now instead

Reply:
Ah never mind, I thought we were duplicating work in the UpdateChecker since both `compile_updates_for(dependency, dependency_files)` and `create_change_for(lead_dependency, updated_dependencies, dependency_files)` call out to it.

I think it would make sense to let the DependencyChangeBuilder use the UpdateChecker to fetch `updated_dependencies` rather than having to pass them in separately in `create_change_for`

Reply:
Yeah, I've thought a bit about that but a lot of the operation-specific logic differs in terms of how the `UpdateChecker` is tested before we decide to proceed to actually creating a dependency change.

I think that will end up being the main source of logic that is different for each operation.